miden-tx/tests/test_miden_faucet_contract.rs

```diff
@@ -0,0 +1,433 @@
+        Felt::new(1),
+    )
+}
+
+#[test]
+fn test_faucet_contract_mint_fungible_asset_succeeds() {
+    let faucet_account = get_faucet_account_with_max_suppply(200);
+
+    // CONSTRUCT AND EXECUTE TX (Success)
+    // --------------------------------------------------------------------------------------------
+    let data_store = MockDataStore::with_existing(Some(faucet_account.clone()), Some(vec![]));
+
+    let mut executor = TransactionExecutor::new(data_store.clone());
+    executor.load_account(faucet_account.id()).unwrap();
+
+    let block_ref = data_store.block_header.block_num().as_int() as u32;
+    let note_origins = data_store
+        .notes
+        .iter()
+        .map(|note| note.proof().as_ref().unwrap().origin().clone())
+        .collect::<Vec<_>>();
+
+    let recipient = [Felt::new(0), Felt::new(1), Felt::new(2), Felt::new(3)];
+    let tag = Felt::new(4);
+    let amount = Felt::new(100);
+
+    let tx_script = ProgramAst::parse(
+        format!(
+            "
+            use.miden::faucets::basic->faucet
+            use.miden::eoa::basic->auth_tx
+    
+            begin
+
+                push.{recipient}
+                push.{tag}
+                push.{amount}
+                call.faucet::distribute
+                
+                call.auth_tx::auth_tx_rpo_falcon512
+                dropw dropw
+
+            end
+            ",
+            recipient = prepare_word(&recipient),
+            tag = tag,
+            amount = amount,
+        )
+        .as_str(),
+    )
+    .unwrap();
+
+    // Execute the transaction and get the witness
+    let transaction_result = executor
+        .execute_transaction(faucet_account.id(), block_ref, &note_origins, Some(tx_script))
+        .unwrap();
+
+    let fungible_asset_1: Asset = FungibleAsset::new(faucet_account.id(), 100).unwrap().into();
+
+    let expected_note = NoteStub::new(
+        recipient.into(),
+        NoteVault::new(&[fungible_asset_1]).unwrap(),
+        NoteMetadata::new(faucet_account.id(), Felt::new(4), Felt::new(1)),
+    )
+    .unwrap();
+
+    assert!(transaction_result.created_notes().notes()[0].recipient() == expected_note.recipient());
+    assert!(transaction_result.created_notes().notes()[0].vault() == expected_note.vault());
+    assert!(transaction_result.created_notes().notes()[0].metadata() == expected_note.metadata());
+}
+
+#[test]
+fn test_faucet_contract_mint_fungible_asset_fails_exceeds_max_supply() {
+    let faucet_account_id = AccountId::try_from(ACCOUNT_ID_FUNGIBLE_FAUCET_ON_CHAIN).unwrap();
+    let faucet_account_code_src = include_str!("../../miden-lib/asm/faucets/basic.masm");
+    let faucet_account_code_ast = ModuleAst::parse(faucet_account_code_src).unwrap();
+    let mut account_assembler = assembler();
+
+    let faucet_account_code = AccountCode::new(
+        faucet_account_id,
+        faucet_account_code_ast.clone(),
+        &mut account_assembler,
+    )
+    .unwrap();
```

Comment:
Here and in other tests, we can probably use `get_faucet_account_with_max_suppply()` function.